front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -0,0 +1,532 @@
+  DataSourceViewType,
+  LightContentNode,
+  LightWorkspaceType,
+  PlanType,
+  PostDataSourceDocumentRequestBody,
+} from "@dust-tt/types";
+import { MAX_FILE_LENGTH, MAX_FILE_SIZES } from "@dust-tt/types";
+import {
+  BIG_FILE_SIZE,
+  Err,
+  isSlugified,
+  parseAndStringifyCsv,
+} from "@dust-tt/types";
+import React, { useContext, useEffect, useRef, useState } from "react";
+
+import { DocumentLimitPopup } from "@app/components/data_source/DocumentLimitPopup";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { handleFileUploadToText } from "@app/lib/client/handle_file_upload";
+import { useTable } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+
+interface DocumentOrTableUploadOrEditModalProps {
+  dataSourceView: DataSourceViewType;
+  contentNode?: LightContentNode;
+  isOpen: boolean;
+  onClose: (save: boolean) => void;
+  owner: LightWorkspaceType;
+  plan: PlanType;
+}
+
+interface TableOrDocument {
+  name: string;
+  description: string;
+  file: File | null;
+  text: string;
+  tags: string[];
+  sourceUrl: string;
+}
+
+export function DocumentOrTableUploadOrEditModal({
+  dataSourceView,
+  contentNode,
+  isOpen,
+  onClose,
+  owner,
+  plan,
+}: DocumentOrTableUploadOrEditModalProps) {
+  const sendNotification = useContext(SendNotificationsContext);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const [tableOrDoc, setTableOrDoc] = useState<TableOrDocument>({
+    name: "",
+    description: "",
+    file: null,
+    text: "",
+    tags: [],
+    sourceUrl: "",
+  });
+
+  const [uploading, setUploading] = useState(false);
+  const [isBigFile, setIsBigFile] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [developerOptionsVisible, setDeveloperOptionsVisible] = useState(false);
+
+  const isTable = contentNode?.type === "database";
+  const initialId = contentNode?.internalId;
+
+  const { table } = useTable({
+    workspaceId: owner.sId,
+    dataSourceName: dataSourceView.dataSource.name,
+    tableId: isTable ? initialId ?? null : null,
+  });
+
+  const resetTableOrDoc = () => {
+    setTableOrDoc({
+      name: "",
+      description: "",
+      file: null,
+      text: "",
+      tags: [],
+      sourceUrl: "",
+    });
+  };
+
+  // TODO(GROUPS_UI) replace endpoint https://github.com/dust-tt/dust/issues/6921
+  useEffect(() => {
+    const fetchData = async () => {
+      if (!initialId) {
+        resetTableOrDoc();
+        return;
+      }
+      setIsLoading(true);
+      if (isTable && table) {
+        setTableOrDoc((prev) => ({
+          ...prev,
+          name: table.name,
+          description: table.description,
+        }));
+      } else {
+        setUploading(true);
+        try {
+          const res = await fetch(
+            `/api/w/${owner.sId}/data_sources/${
+              dataSourceView.dataSource.name
+            }/documents/${encodeURIComponent(initialId)}`
+          );
```

Comment:
maybe we could move the code in a hook like for useTable ?

Reply:
Definitely 😎